inst/include/distributions/functors/lognormal_lpdf.hpp

```diff
@@ -1,74 +1,93 @@
-#ifndef LOGLogNormal_LPDF 
-#define LOGLogNormal_LPDF 
+#ifndef LOGNORMAL_LPDF
+#define LOGNORMAL_LPDF
 
 #include "density_components_base.hpp"
-#include "../common/fims_vector.hpp"
-#include "../common/def.hpp"
+#include "../../common/fims_vector.hpp"
+#include "../../common/def.hpp"
 
-namespace fims_distributions {
-/**
- * LogNormal Negative Log-Likelihood
- */
-template<typename Type>
-struct LogNormalLPDF : public DensityComponentBase<Type> {
-    fims::Vector<Type> log_sd;
-    fims::Vector<Type> mu;
-    fims::Vector<Type> sd;
-    Type nll = 0.0;
-    //data_indicator<tmbutils::vector<Type> , Type> keep;
-   
+namespace fims_distributions
+{
+    /**
+     * LogNormal Negative Log-Likelihood
+     */
+    template <typename Type>
+    struct LogNormalLPDF : public DensityComponentBase<Type>
+    {
+        fims::Vector<Type> log_sd;
+        fims::Vector<Type> mu;
+        fims::Vector<Type> sd;
+        std::vector<bool> is_na;
+        #ifdef TMB_MODEL
+        ::objective_function<Type> *of;
+        #endif
+        Type nll = 0.0;
+        // data_indicator<tmbutils::vector<Type> , Type> keep;
 
-    LogNormalLPDF() : DistributionComponentBase<Type>() {
+        LogNormalLPDF() : DensityComponentBase<Type>()
+        {
+        }
 
-    }
+        virtual ~LogNormalLPDF() {}
 
-    virtual ~LogNormalLPDF() {}
+        virtual const Type evaluate(const bool &do_log)
```

Comment:
I actually don't think we can use do_log=FALSE here b/c later we sum up which won't make sense for negative densities. What is the use case here? Conversion back to natural space, when necessary, should take place outside of this function I think.

Reply:
will add task to next sprint